### PR TITLE
Fix flaky Local SDK test

### PR DESCRIPTION
**Pull request by aLekSer**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature

/kind hotfix

**What this PR does / Why we need it**:
Fix LocalSDK test which is flaky.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

There is still a timeout possible here:
https://github.com/googleforgames/agones/blob/d30668c661119b1121bf2a43d20949e8d5d13783/pkg/sdkserver/localsdk_test.go#L281

### CODE REVIEW

File: pkg/sdkserver/localsdk_test.go [hunk 1]
```diff
@@ -125,6 +125,8 @@ func TestLocalSDKServerSetLabel(t *testing.T) {
 	}
 
 	for k, v := range fixtures {
+		k := k
```
[the remark this answers is not in the document]
Comment (aLekSer) [reply pkg/sdkserver/localsdk_test.go:128]: yes, I will do (there is a `gocritic` linter article about this), btw on the multiple runs still get some errors. Will update the test one more time

**Comment by aLekSer** [comment]
Strange enough I have received error here, something wrong with `runtime.Features`:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1e977ec]

goroutine 541 [running]:
testing.tRunner.func1(0xc00040b300)
	/usr/local/Cellar/go/1.13.1/libexec/src/testing/testing.go:874 +0x3a3
panic(0x1ffa7e0, 0x2e23980)
	/usr/local/Cellar/go/1.13.1/libexec/src/runtime/panic.go:679 +0x1b2
agones.dev/agones/pkg/sdkserver.TestLocalSDKServerPlayerConnectAndDisconnect.func2(0xc00040b300)
	/Users/alexander.apalikov/go/src/github.com/agones/agones/pkg/sdkserver/localsdk_test.go:460 +0x61c
testing.tRunner(0xc00040b300, 0xc0008ea210)
	/usr/local/Cellar/go/1.13.1/libexec/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/Cellar/go/1.13.1/libexec/src/testing/testing.go:960 +0x350
exit status 2
FAIL	agones.dev/agones/pkg/sdkserver	13.300s
```

**Comment by markmandel** [comment]
> Strange enough I have received error here, something wrong with runtime.Features:

Seems like the feature flag got flipped mid test? Is there a test somewhere that's not been locked appropriately I wonder.

I wonder if we want to add some debug logging [here](https://github.com/googleforgames/agones/blob/76b69bc8ffbf3bbc589dd6bd9b38da1e59ba8ced/pkg/util/runtime/features.go#L86) and [here](https://github.com/googleforgames/agones/blob/76b69bc8ffbf3bbc589dd6bd9b38da1e59ba8ced/pkg/util/runtime/features.go#L123) to help with picking this up?

File: pkg/sdkserver/localsdk_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (aLekSer) [reply pkg/sdkserver/localsdk_test.go:128]: It was not `gocritic`, but https://github.com/kyoh86/scopelint and it is obsolete, with two successors.

**Comment by aLekSer** [comment]
Will add this comment here:
```
	runtime.FeatureTestMutex.Lock()
	defer runtime.FeatureTestMutex.Unlock()
	for k, v := range fixtures {
		k := k
		v := v
		t.Run(k, func(t *testing.T) {
			t.Parallel()
			runtime.FeatureTestMutex.Lock()
			defer runtime.FeatureTestMutex.Unlock()
...
```
This one above would not have a deadlock and without t.Parallel()  it has. So nested parallel tests could use different variables set.

**Comment by aLekSer** [comment]
Run all tests in `pkg/sdkserver/localsdk_test.go` for 2k times.

File: pkg/sdkserver/localsdk_test.go [hunk 2]
```diff
@@ -380,9 +393,10 @@ func TestLocalSDKServerPlayerConnectAndDisconnect(t *testing.T) {
 	}
 
 	for k, v := range fixtures {
+		// pin variables here, see https://github.com/kyoh86/scopelint for the details
+		k := k
+		v := v
 		t.Run(k, func(t *testing.T) {
-			t.Parallel()
```
Comment (aLekSer) [inline pkg/sdkserver/localsdk_test.go:384]: This line leads to unpredictable behaviour. I assume we can add this in a separate PR.

**Comment by aLekSer** [comment]
One rare failed test still exist (1 out of 144 tests):
```
--- FAIL: TestLocal (0.00s)
    localsdk_test.go:67: 
        	Error Trace:	localsdk_test.go:67
        	Error:      	Not equal: 
        	            	expected: &sdk.GameServer_ObjectMeta{Name:"local", Namespace:"default", Uid:"1234", ResourceVersion:"v1", Generation:1, CreationTimestamp:1590521272, DeletionTimestamp:0, Annotations:map[string]string{"annotation":"true"}, Labels:map[string]string{"islocal":"true"}, XXX_NoUnkeyedLiteral:struct {}{}, XXX_unrecognized:[]uint8(nil), XXX_sizecache:0}
        	            	actual  : &sdk.GameServer_ObjectMeta{Name:"local", Namespace:"default", Uid:"1234", ResourceVersion:"v1", Generation:1, CreationTimestamp:1590521271, DeletionTimestamp:0, Annotations:map[string]string{"annotation":"true"}, Labels:map[string]string{"islocal":"true"}, XXX_NoUnkeyedLiteral:struct {}{}, XXX_unrecognized:[]uint8(nil), XXX_sizecache:0}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,2 @@
        	            	-(*sdk.GameServer_ObjectMeta)(name:"local" namespace:"default" uid:"1234" resource_version:"v1" generation:1 creation_timestamp:1590521272 annotations:<key:"annotation" value:"true" > labels:<key:"islocal" value:"true" > )
        	            	+(*sdk.GameServer_ObjectMeta)(name:"local" namespace:"default" uid:"1234" resource_version:"v1" generation:1 creation_timestamp:1590521271 annotations:<key:"annotation" value:"true" > labels:<key:"islocal" value:"true" > )
        	            	 
        	Test:       	TestLocal
```
And `localsdk_test.go:282` could timeout locally at least.

**Pull request merged**